### PR TITLE
various improvements

### PR DESCRIPTION
**Pull request by chriddyp**
cc @cldougl and @scjody and @BRONSOLO and @plotly/dash 

various improvements to the dash sample.
fixes https://github.com/plotly/streambed/issues/10749
fixes https://github.com/plotly/streambed/issues/10750
fixes https://github.com/plotly/streambed/issues/10751

and includes a few other improvements that i talked to @cldougl about today.

best to read this one commit-by-commit

### CODE REVIEW

**Comment by scjody** [comment]
I'll review this this evening or early Monday (assuming it's ready for review...)

Are you planning on doing something similar for `dash-auth`? Or is it already verbose to the max?

File: auth.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import os
+from textwrap import dedent
+import urllib3
+
+import config
+
+########################################################################
+# This file provides an interface to the `plotly_auth` library
+# You do not need to edit this file
+########################################################################
```
[the remark this answers is not in the document]
Comment (chriddyp) [reply auth.py]: Yeah, I'm just so afraid of users missing my comments without something bold like this and then editing the file or asking us if they need to edit the file.

Maybe just adding `**not**`? Any other suggestions for emphasizing this?
```
# This file provides an interface to the `plotly_auth` library
# You do **not** need to edit this file
```

[the remark this answers is not in the document]
Comment (scjody) [reply auth.py]: Looks like this is accepted:
```
# # # # # # # # #
```

File: auth.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            PLOTLY_API_KEY=os.environ['PLOTLY_API_KEY'],
+            PLOTLY_DOMAIN=os.environ['PLOTLY_DOMAIN'],
+            PLOTLY_API_DOMAIN=os.environ['PLOTLY_API_DOMAIN'],
+            PLOTLY_DASH_DOMAIN=config.PLOTLY_DASH_DOMAIN,
+            PLOTLY_SSL_VERIFICATION=os.environ['PLOTLY_SSL_VERIFICATION']
+        )))
+    if config.DASH_APP_PRIVACY == 'private':
+        # Checks if running inside Plotly On-Premise environment
+        if 'DYNO' in os.environ:
+            if config.PATH_BASED_ROUTING:
```
Comment (scjody) [inline auth.py]: https://github.com/plotly/streambed/issues/10763 was introduced here.

**Pull request merged**